src/main/java/com/ita/if103java/ims/service/impl/AccountServiceImpl.java

```diff
@@ -40,6 +57,12 @@ public AccountDto view(Long id) {
 
     @Override
     public boolean delete(Long id) {
+        Event event = new Event();
+        event.setMessage("Account was deleted.");
+        event.setAccountId(id);
+        event.setAuthorId(accountDao.findById(id).getAdminId());
+        event.setName(EventName.ORG_DELETED);
+        eventService.create(event);
```

Comment:
Please create method createEvent in this class and pass there only those arguments which changes.
`event.setAuthorId(accountDao.findById(id).getAdminId());`
Can be select user from the @AuthenticationPrincipal as in methods above